### PR TITLE
Ruby: Add consistency query for extraction errors

### PR DESCRIPTION
**Pull request by hvitved**
Revealed a few syntax errors that I have fixed, and some where I don't know whether it's really a bug in the Tree-sitter grammar (where I have simply accepted the inconsistency).

### CODE REVIEW

File: ruby/ql/test/library-tests/ast/CONSISTENCY/AstConsistency.expected [hunk 1]
```diff
@@ -0,0 +1,2 @@
+extractionError
+| calls/calls.rb:8:1:8:2 | A parse error occurred. Check the syntax of the file. If the file is invalid, correct the error or exclude the file from analysis. |
```
Comment (aibaars) [inline ruby/ql/test/library-tests/ast/CONSISTENCY/AstConsistency.expected:2]: I think the expression on line 8 is just nonsensical, probably best remove it unless the `ruby` interpreter accepts it.

File: ruby/ql/test/library-tests/frameworks/action_view/CONSISTENCY/AstConsistency.expected [hunk 2]
```diff
@@ -0,0 +1,2 @@
+extractionError
+| app/views/foo/bars/show.html.erb:10:8:11:7 | A parse error occurred. Check the syntax of the file. If the file is invalid, correct the error or exclude the file from analysis. |
```
Comment (aibaars) [inline ruby/ql/test/library-tests/frameworks/action_view/CONSISTENCY/AstConsistency.expected:2]: I think a `%>` is missing after the `do` .

**Review by aibaars — APPROVED** [review]
Looks good to me, let's fix the syntax errors in our test files. Feel free to do the fixes in a follow-up PR.

File: ruby/ql/test/library-tests/dataflow/flow-summaries/CONSISTENCY/AstConsistency.expected [hunk 3]
```diff
@@ -0,0 +1,3 @@
+extractionError
+| semantics.rb:36:18:36:23 | A parse error occurred. Check the syntax of the file. If the file is invalid, correct the error or exclude the file from analysis. |
```
[the remark this answers is not in the document]
Comment (hvitved) [reply ruby/ql/test/library-tests/dataflow/flow-summaries/CONSISTENCY/AstConsistency.expected]: Thanks; fixed now.

File: ruby/ql/test/library-tests/ast/CONSISTENCY/AstConsistency.expected (hunk 1 above)
Comment (hvitved) [reply ruby/ql/test/library-tests/ast/CONSISTENCY/AstConsistency.expected]: Thanks; fixed now.

File: ruby/ql/test/library-tests/frameworks/action_view/CONSISTENCY/AstConsistency.expected (hunk 2 above)
Comment (hvitved) [reply ruby/ql/test/library-tests/frameworks/action_view/CONSISTENCY/AstConsistency.expected]: Even after adding `%>` there is still a parse error.

Comment (aibaars) [reply ruby/ql/test/library-tests/frameworks/action_view/CONSISTENCY/AstConsistency.expected]: Yeah the `key = [:display_text, :safe_text] do` phrase does not seem to make any sense. Perhaps they intended `for key in [:display_text, :safe_text] do` or similiar.

Comment (hvitved) [reply ruby/ql/test/library-tests/frameworks/action_view/CONSISTENCY/AstConsistency.expected]: Right. Fixed as well.

**Comment by hvitved** [comment]
@aibaars : Can you reapprove, please?

**Review by aibaars — APPROVED** [review]

**Pull request merged**